### PR TITLE
gmp: update to 6.3.0

### PR DESCRIPTION
**Pull request by szhorvat**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 13.5.1 22G90 arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by szhorvat** [comment]
Ran the tests both for the default and `+native` variants, they passed.

----

Removed the patch since it appeared to be for Apple Silicon support, but [the changelog](https://gmplib.org/gmp6.3) indicates that 6.3.0 provides this support.

----

With `+native`, there's a warning from the Portfile:

> No processor dependent assembly code being used. gmp might be slower.

I am not sure if this is still valid.

----

File: devel/gmp/Portfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ PortGroup               muniversal       1.1
 PortGroup               xcode_workaround 1.0
 
 name                    gmp
-version                 6.2.1
+version                 6.3.0
 revision                1
```
Comment (reneeotten) [inline devel/gmp/Portfile:9]: the "revision" field should be set to "0" when updating the version

**Review by reneeotten — CHANGES_REQUESTED** [review]
also the "platforms" line is the default since MacPorts v1.7 so that line can/should be removed.

**Comment by MarcusCalhoun-Lopez** [comment]
I am afraid there are two issues with this pull request other than the one pointed out by @reneeotten.
The first is minor.
`triplet.cpu.arm64` should be changed to `applem1` (see https://gmplib.org/repo/gmp/rev/0d19622b27a9).

The second is that version 6.3.0 does not work on (very) old systems.
The problem seems to revolve around setting the correct language standard in `CC_FOR_BUILD` and/or `CFLAGS_FOR_BUILD`, but I still haven't had quite enough time to find a proper fix.

**Comment by szhorvat** [comment]
At this point, I would hand this over to you @MarcusCalhoun-Lopez. I do not have the possibility to test on old systems. I'll close this PR since the changes are so minor that there's no reason to build on top of it (but if you want to, just reopen).

**Pull request closed without merging**